frontend/src/components/JsonViewer/json-box.vue

```diff
@@ -0,0 +1,160 @@
+<script>
+import JsonString from "./types/json-string.vue";
+import JsonUndefined from "./types/json-undefined.vue";
+import JsonNumber from "./types/json-number.vue";
+import JsonBoolean from "./types/json-boolean.vue";
+import JsonObject from "./types/json-object.vue";
+import JsonArray from "./types/json-array.vue";
+import JsonFunction from "./types/json-function.vue";
+import JsonDate from "./types/json-date.vue";
+import JsonRegexp from "./types/json-regexp.vue";
+import { h } from "vue";
+export default {
+  name: "JsonBox",
+  inject: ["expandDepth", "keyClick"],
+  props: {
+    value: {
+      type: [Object, Array, String, Number, Boolean, Function, Date],
+      default: null,
+    },
+    keyName: {
+      type: String,
+      default: "",
+    },
+    sort: Boolean,
+    depth: {
+      type: Number,
+      default: 0,
+    },
+    previewMode: Boolean,
+  },
+  data() {
+    return {
+      expand: false,
+    };
+  },
+  mounted() {
+    // this.expand = this.previewMode || (this.depth >= this.expandDepth ? false : true);
+  },
+  methods: {
+    toggle() {
+      this.expand = !this.expand;
+
+      try {
+        this.$el.dispatchEvent(new Event("resized"));
+      } catch (e) {
+        // handle IE not supporting Event constructor
+        var evt = document.createEvent("Event");
+        evt.initEvent("resized", true, false);
+        this.$el.dispatchEvent(evt);
+      }
+    },
+  },
+  render() {
+    let elements = [];
+    let dataType;
+
+    if (this.value === null || this.value === undefined) {
+      dataType = JsonUndefined;
+    } else if (Array.isArray(this.value)) {
+      dataType = JsonArray;
+    } else if (Object.prototype.toString.call(this.value) === "[object Date]") {
+      dataType = JsonDate;
+    } else if (typeof this.value === "object") {
+      dataType = JsonObject;
+    } else if (typeof this.value === "number") {
+      dataType = JsonNumber;
+    } else if (typeof this.value === "string") {
+      dataType = JsonString;
+    } else if (typeof this.value === "boolean") {
+      dataType = JsonBoolean;
+    } else if (typeof this.value === "function") {
+      dataType = JsonFunction;
+    }
+    if (this.value.constructor === RegExp) {
+      // console.log("type", this.value.constructor === RegExp);
+      // this.value=this.value.toString()
+      dataType = JsonRegexp;
+    }
+    const complex =
+      this.keyName &&
+      this.value &&
+      (Array.isArray(this.value) ||
+        (typeof this.value === "object" &&
+          Object.prototype.toString.call(this.value) !== "[object Date]"));
+
+    if (!this.previewMode && complex) {
```

Comment:
@bradleySuira could you please add some comments about what is this doing and why is it needed?

Reply:
added